be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1622,8 +1618,49 @@ Status SegmentIterator::_read_columns_by_index(uint32_t nrows_read_limit, uint32
         }
 
         _split_row_ranges.emplace_back(std::pair {range_from, range_to});
-        // if _opts.read_orderby_key_reverse is true, only read one range for fast reverse purpose
+
+        _read_range_info.add(range_from, range_to);
     } while (nrows_read < nrows_read_limit && !_opts.read_orderby_key_reverse);
+
+    if (_read_range_info.ranges.empty()) {
+        return Status::OK();
+    }
+
+    if (_opts.runtime_state) {
+        int32_t threshold = _opts.runtime_state->query_options().inverted_index_read_mode_threshold;
```

Comment:
the name inverted_index_read_mode_threshold has inverted index, but the logic is not only for inverted index.